one/src/cbor_value.rs

```diff
@@ -0,0 +1,285 @@
+            Ok(Type::String) => Ok(CborValue::String(decoder.str()?.to_string())),
+            Ok(Type::StringIndef) => Ok(CborValue::Undefined), // TODO: support Type::StringIndef
+            Ok(Type::Array) => {
+                let mut array = Vec::new();
+                for _ in 0..decoder.array().unwrap().unwrap() {
+                    let value = CborValue::next(decoder)?;
+                    array.push(value);
+                }
+                Ok(CborValue::Array(array))
+            }
+            Ok(Type::ArrayIndef) => {
+                let mut array = Vec::new();
+                loop {
+                    let value = CborValue::next(decoder)?;
+                    if let CborValue::Break = value {
+                        break;
+                    }
+                    array.push(value)
+                }
+                Ok(CborValue::Array(array))
+            }
+            Ok(Type::Map) => {
+                let mut map = BTreeMap::new();
+                for _ in 0..decoder.map().unwrap().unwrap() {
+                    let key = CborValue::next(decoder)?;
+                    let value = CborValue::next(decoder)?;
+                    map.insert(key, value);
+                }
+                Ok(CborValue::Map(map))
+            }
+            Ok(Type::MapIndef) => Ok(CborValue::Undefined), // TODO: support Type::MapIndef
+            Ok(Type::Tag) => {
+                let tag: Tag = decoder.tag()?;
+                let value = CborValue::next(decoder)?;
+                Ok(CborValue::Tag((tag, Box::new(value))))
+            }
+            Ok(Type::Break) => Ok(CborValue::Break),
+            Ok(Type::Unknown(additional)) => Ok(CborValue::Unknown(additional)),
+            Err(e) => Err(e.into()),
+        }
+    }
+
+    /// Accesses a deeply nested CborValue when the path keys are all strings.
+    pub fn path(&self, parts: &[&str]) -> CborValue {
+        match parts.split_first() {
+            None => self.clone(), // if there are no parts this is thing at the path.
+            Some((first, rest)) => {
+                let CborValue::Map(map) = &self else {
+                    return CborValue::Undefined; // if self is not a map there is no thing at the path.
+                };
+                let Some(next) = map.get(&first.to_string().into()) else {
+                    return CborValue::Undefined; // if the next part is not in the map there is no thing at the path.
+                };
+                next.path(rest)
+            }
+        }
+    }
+
+    /// Accesses the underlying CborValue for a specified tag type.
+    pub fn untag(&self, tag: u64) -> Option<&CborValue> {
+        if let CborValue::Tag((Tag::Unassigned(inner_tag), boxed_value)) = &self {
+            if inner_tag == &tag {
+                return Some(boxed_value);
+            }
+        }
+        None
+    }
+
+    /// get a printable string of the type_name
+    pub fn type_name(&self) -> String {
```

Comment:
could this just return &'static str? I don't think there's a good way to use std::fmt::Display for this behavior, but I don't think we need to allocate the response.

Reply:
I will return `&'static str`